src/coreclr/src/vm/amd64/asmhelpers.S

```diff
@@ -9,141 +9,180 @@
+
+# Pad the frame size by 0x8 so when the xmm0 and xmm1 register store/restore happens
+# we can align to 16 and be guaranteed to not exceed the frame size
+.equ STACK_FUDGE_FACTOR, 0x8
+
+# Space to keep xmm0 and xmm1
+.equ SIZEOF_FP_ARG_SPILL, 0x10*2
+
+.equ OFFSETOF_FP_ARG_SPILL, SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA
+
+# SIZEOF_STACK_FRAME is how many bytes we reserve in our ELT helpers below
+# There are three components, the first is space for profiler platform specific
+# data struct that we spill the general purpose registers to, then space to
+# spill xmm0 and xmm1, then finally 8 bytes of padding to ensure that the xmm
+# register reads/writes are aligned on 16 bytes.
+.equ SIZEOF_STACK_FRAME, SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA + SIZEOF_FP_ARG_SPILL + STACK_FUDGE_FACTOR
+
+# general purpose registers
+.equ SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA, 0x8*22
+
+# Pad the frame size by 0x8 so when the xmm0 and xmm1 register store/restore happens
+# we can align to 16 and be guaranteed to not exceed the frame size
+.equ STACK_FUDGE_FACTOR, 0x8
+
+# Space to keep xmm0 and xmm1
 .equ SIZEOF_FP_ARG_SPILL, 0x10*2
 
-.equ SIZEOF_STACK_FRAME, SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA + SIZEOF_FP_ARG_SPILL
+.equ OFFSETOF_FP_ARG_SPILL, SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA
+
+# SIZEOF_STACK_FRAME is how many bytes we reserve in our ELT helpers below
+# There are three components, the first is space for profiler platform specific
+# data struct that we spill the general purpose registers to, then space to
+# spill xmm0 and xmm1, then finally 8 bytes of padding to ensure that the xmm
+# register reads/writes are aligned on 16 bytes.
+.equ SIZEOF_STACK_FRAME, SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA + SIZEOF_FP_ARG_SPILL + STACK_FUDGE_FACTOR
 
 .equ PROFILE_ENTER, 0x1
 .equ PROFILE_LEAVE, 0x2
 .equ PROFILE_TAILCALL, 0x4
 
-// ***********************************************************
-//   NOTE:
-//
-//   Register preservation scheme:
-//
-//       Preserved:
-//           - all non-volatile registers
-//           - rax, rdx
-//           - xmm0, xmm1
-//
-//       Not Preserved:
-//           - integer argument registers (rcx, rdx, r8, r9)
-//           - floating point argument registers (xmm1-3)
-//           - volatile integer registers (r10, r11)
-//           - volatile floating point registers (xmm4-5)
-//           - upper halves of ymm registers on AVX (which are volatile)
-//
-// ***********************************************************
-
-// EXTERN_C void ProfileEnterNaked(FunctionIDOrClientID functionIDOrClientID, size_t profiledRsp);
-// <NOTE>
-//
-// </NOTE>
+# ***********************************************************
+#   NOTE:
+#
+#   Register preservation scheme:
+#
+#       Preserved:
+#           - all non-volatile registers
+#           - rax, rdx
+#           - xmm0, xmm1
+#
+#       Not Preserved:
+#           - integer argument registers (rcx, rdx, r8, r9)
```

Comment:
It predates your change but may as well fix it with other inconsistencies, the code does appear to preserve rcx, rdx, r8, and r9. The comment is also self-inconsistent claiming above that rdx was preserved + claiming it is not preserved here.